github-stats/fetcher/deploy

```diff
@@ -0,0 +1,48 @@
+#!/bin/sh
+
+# Copyright 2016 The Kubernetes Authors All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+exec 3>&1
+
+# Print a line along with dates that goes to user output (not log file)
+# Args are processed like with echo(1)
+print_line() {
+  echo $(date +"%F-%H %T.%N"): "$@" >&3
+}
+
+# Print what a command will do and run the command. Report if it fails.
+# usage: run_command "What I'm doing" how_im_doing_it with parameters
+run_command() {
+  what=$1
+  shift
+  print_line $what
+  "$@" || {
+    print_line Failed. You can have a look at \"deploy.log\"
+    exit 3
```

Comment:
You could do something like just return 3 here and then below do
```
if run_command "Running tests" go test ./; then
  print_line "Finished successfully"
else
  print_line "Finished with error"
fi
```

That way you don't have to unset errexit.